library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -2418,40 +2426,72 @@ public boolean onTouchEvent(MotionEvent event) {
-
-        if (mCurrentFlingDirection != Direction.NONE) {
-            // snap to nearest day
-            leftDays = Math.round(leftDays);
-        } else if (mCurrentScrollDirection == Direction.LEFT) {
-            // snap to last day
-            leftDays = Math.floor(leftDays);
-        } else if (mCurrentScrollDirection == Direction.RIGHT) {
-            // snap to next day
-            leftDays = Math.ceil(leftDays);
-        } else {
-            // snap to nearest day
-            leftDays = Math.round(leftDays);
-        }
+        if(!mIsCustomScrollableAmountsOfDaysEnabled) {
+            double leftDays = mCurrentOrigin.x / (mWidthPerDay + mColumnGap);
 
-        int nearestOrigin = (int) (mCurrentOrigin.x - leftDays * (mWidthPerDay + mColumnGap));
-        boolean mayScrollHorizontal = mCurrentOrigin.x - nearestOrigin < getXMaxLimit()
-                && mCurrentOrigin.x - nearestOrigin > getXMinLimit();
+            if (mCurrentFlingDirection != Direction.NONE) {
+                // snap to nearest day
+                leftDays = Math.round(leftDays);
+            } else if (mCurrentScrollDirection == Direction.LEFT) {
+                // snap to last day
+                leftDays = Math.floor(leftDays);
+            } else if (mCurrentScrollDirection == Direction.RIGHT) {
+                // snap to next day
+                leftDays = Math.ceil(leftDays);
+            } else {
+                // snap to nearest day
+                leftDays = Math.round(leftDays);
+            }
 
-        if (mayScrollHorizontal) {
-            mScroller.startScroll((int) mCurrentOrigin.x, (int) mCurrentOrigin.y, - nearestOrigin, 0);
-            ViewCompat.postInvalidateOnAnimation(WeekView.this);
-        }
+            int nearestOrigin = (int) (mCurrentOrigin.x - leftDays * (mWidthPerDay + mColumnGap));
+            boolean mayScrollHorizontal = mCurrentOrigin.x - nearestOrigin < getXMaxLimit()
+                    && mCurrentOrigin.x - nearestOrigin > getXMinLimit();
+
+            if (mayScrollHorizontal) {
+                mScroller.startScroll((int) mCurrentOrigin.x, (int) mCurrentOrigin.y, -nearestOrigin, 0);
+                ViewCompat.postInvalidateOnAnimation(WeekView.this);
+            }
+
+            if (nearestOrigin != 0 && mayScrollHorizontal) {
+                // Stop current animation.
+                mScroller.forceFinished(true);
+                // Snap to date.
+                mScroller.startScroll((int) mCurrentOrigin.x, (int) mCurrentOrigin.y, -nearestOrigin, 0, (int) (Math.abs(nearestOrigin) / mWidthPerDay * mScrollDuration));
+                ViewCompat.postInvalidateOnAnimation(WeekView.this);
+            }
+            // Reset scrolling and fling direction.
+            mCurrentScrollDirection = mCurrentFlingDirection = Direction.NONE;
+        }else {
+            sizeOfScrollableView = (mWidthPerDay + mColumnGap) * getNumberOfVisibleDays();
```

Comment:
There is some code duplication in the if part and in the else part. It would be best, if that part can be executed outside of the if else. Especially the `mScroller.startScroll` etc

Reply:
I could move outside : 

// Reset scrolling and fling direction.
mCurrentScrollDirection = mCurrentFlingDirection = Direction.NONE;

But the rest of the code is not really a duplicate part :/

Reply:
there are a lot of lines that are equal but have different values for the parameters, you can probably create variables for those parameters, set these variables in the if and else and place a lot of code after the if else